components/dialog/stories/dialog.stories.js

```diff
@@ -265,20 +264,66 @@ WithScroll.args = {
 	}
 };
 WithScroll.storyName = "Scrollable";
-WithScroll.tags = ["is-hidden-story", "!dev"];
+WithScroll.tags = ["!dev"];
 WithScroll.parameters = {
 	chromatic: { disableSnapshot: true },
 };
 
-/* TODO: once this gets rebased with `main`, make sure to use the fullscreen and fullscreenTakeover test templates
-that are imported above, but commented out.
- */
 /**
- * The full screen variant shows a large dialog background, only revealing a small portion of the page around the outside of the dialog, behind an overlay. The size of the dialog varies with the size of the screen, in both width and height.
+ * The fullscreen variant shows a large dialog background, only revealing a small portion of the page around the outside of the dialog, behind an overlay. The size of the dialog varies with the size of the screen, in both width and height.
+ *
+ * Implementations may swap out the extra header content and body content for other components, like the [steplist](/docs/components-steplist--docs) and [table](/docs/components-table--docs) seen in this example. Content in this area will be centered.
```

Comment:
What are your thoughts on these changes- allowing full-fledged components to be rendered in the header content & body areas? That's how it is encouraged in the [Figma files](https://www.figma.com/design/Mngz9H7WZLbrCvGQf3GnsY/S2-%2F-Desktop?node-id=61935-5399&m=dev), so I think it makes sense to at least _show_ the possibility in our documentation. 

<img width="975" alt="Screenshot 2024-12-10 at 12 56 20 PM" src="https://github.com/user-attachments/assets/4c1c040a-80b1-434e-bde9-c41144bcdef9">

Is it ok to have in this PR? I tried to wrap all of the "slot-related" changes to a single commit, so it could be really easy to transfer to another branch if we feel like it should be separate.